src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -148,52 +178,66 @@ class Proxied<T> {
   }
 }
 
+async function resolveDeveloperNodeModule(
+  frb: FunctionsRuntimeBundle,
+  pkg: PackageJSON,
+  name: string
+): Promise<ModuleResolution> {
+  const dependencies = pkg.dependencies;
+  const devDependencies = pkg.devDependencies;
+  const isInPackageJSON = dependencies[name] || devDependencies[name];
+
+  // If there's no reference to the module in their package.json, prompt them to install it
+  if (!isInPackageJSON) {
+    return { declared: false, installed: false };
+  }
+
+  // Once we know it's in the package.json, make sure it's actually `npm install`ed
+  const modResolution = await requireResolveAsync(name, { paths: [frb.cwd] }).catch(NoOp);
+
+  if (!modResolution) {
+    return { declared: true, installed: false };
+  }
+
+  const modPackageJSON = require(path.join(findModuleRoot(name, modResolution), "package.json"));
+
+  return {
+    declared: true,
+    installed: true,
+    version: modPackageJSON.version,
+  };
+}
+
 async function verifyDeveloperNodeModules(frb: FunctionsRuntimeBundle): Promise<boolean> {
-  let pkg;
-  try {
-    pkg = require(`${frb.cwd}/package.json`);
-  } catch (err) {
+  let pkg = requirePackageJson(frb);
+  if (!pkg) {
     new EmulatorLog("SYSTEM", "missing-package-json", "").log();
     return false;
   }
 
   const modBundles = [
-    { name: "firebase-admin", isDev: false, minVersion: 7 },
-    { name: "firebase-functions", isDev: false, minVersion: 2 },
+    { name: "firebase-admin", isDev: false, minVersion: 8 },
```

Comment:
This could be considered a breaking change.  IMO we should consider the emulators in "beta" like we do with some of our SDKs and not make firebase-tools take major version changes all the time as we try to get things right.